ecommerce/src/main/kotlin/io/hhplus/ecommerce/inventory/application/usecase/StockReservationCommandUseCase.kt

```
@@ -98,25 +100,47 @@ class StockReservationCommandUseCase(
-                // 예약 만료 처리
+                // 2. 예약 만료 처리
                 stockReservationDomainService.expireReservation(reservation)
+
+                // 3. 연결된 주문 만료 처리 (중복 방지)
+                reservation.orderId?.let { orderId ->
+                    if (!processedOrderIds.contains(orderId)) {
+                        try {
+                            orderDomainService.expireOrder(orderId)
+                            processedOrderIds.add(orderId)
+                            logger.info("[ReservationExpiry] 주문 만료 처리 완료: orderId=$orderId")
+                        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)